addons/mail/static/src/core/web/chatter.js

```diff
@@ -271,21 +263,53 @@ export class Chatter extends Component {
     /**
      * @param {number} threadId
      * @param {['activities'|'followers'|'attachments'|'messages'|'suggestedRecipients']} requestList
+     * @param {boolean} refresh
+     * Fetch data for the thread according to the request list.
      */
     load(
         threadId = this.props.threadId,
-        requestList = ["followers", "attachments", "messages", "suggestedRecipients"]
+        requestList = ["followers", "attachments", "messages", "suggestedRecipients"],
+        refresh = true
     ) {
         const { threadModel } = this.props;
-        this.state.thread = this.threadService.getThread(threadModel, threadId);
-        this.scrollPosition.model = this.state.thread?.scrollPosition;
+        const thread = this.threadService.getThread(threadModel, threadId);
+        if (refresh) {
+            this.refresh(threadModel, threadId);
+        }
         if (!threadId) {
             return;
         }
         if (this.props.hasActivities && !requestList.includes("activities")) {
             requestList.push("activities");
         }
-        this.threadService.fetchData(this.state.thread, requestList);
+        this.threadService.fetchData(thread, requestList);
+    }
+
+    /**
+     * @param {string} threadModel
+     * @param {number} threadId
+     * Visually refresh the chatter.
+     */
+    refresh(threadModel = this.props.threadModel, threadId = this.props.threadId) {
```

Comment:
I think this method should be called `changeThread` instead, and only be called when necessary.